packages/auth/src/core/user/token_manager.ts

```diff
@@ -73,20 +73,26 @@ export class StsTokenManager {
     );
   }
 
+  updateFromIdToken(idToken: string): void {
+    _assert(idToken.length !== 0, AuthErrorCode.INTERNAL_ERROR);
+    const expiresIn = _tokenExpiresIn(idToken);
+    this.updateTokensAndExpiration(idToken, null, expiresIn);
+  }
+
   async getToken(
     auth: AuthInternal,
     forceRefresh = false
   ): Promise<string | null> {
-    _assert(
-      !this.accessToken || this.refreshToken,
-      auth,
-      AuthErrorCode.TOKEN_EXPIRED
-    );
+    if (!this.accessToken) {
+      _assert(this.refreshToken, auth, AuthErrorCode.TOKEN_EXPIRED);
+    }
 
     if (!forceRefresh && this.accessToken && !this.isExpired) {
       return this.accessToken;
     }
 
+    _assert(this.refreshToken, auth, AuthErrorCode.TOKEN_EXPIRED);
```

Comment:
should this line be here? seems it's cover in the !acessToken assert above

Reply:
Actually, I think the top assertion should be removed. Done!